internal/proxy/fuse.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"syscall"
+
+	"github.com/hanwen/go-fuse/v2/fs"
```

Comment:
How did we land on go-fuse?

Reply:
It's what we used on v1 to gain support for macOS. I see that https://github.com/winfsp/cgofuse supports Windows in addition to macOS, and so I'll replace the implementation here as an experiment. A wider set of supported OS's seems better.

Reply:
I've revisited this decision and here's what I found:

There are four FUSE libraries for Go:

1. bazil.org/fuse -- a solid library, pure Go implementation, but Linux only. 
2. github.com/jacobsa/fuse -- a fork of the library above. Linux and Darwin are supported. 
3. github.com/hanwen/go-fuse -- pure Go implementation, Linx and Darwin are supported.
4. github.com/winfsp/cgofuse -- broad OS support (Linux, macOS, Windows), but requires CGO.

What we want:
1. Broad OS support -- especially Linux and macOS.
2. No CGO -- we compile cross-platform and static binaries. CGO would introduce a lot of complexity into our build process that we'd rather avoid.
3. Well maintained and active project.

By the criteria above, that leaves us with:

1. github.com/jacobsa/fuse and
2. github.com/hanwen/go-fuse

Option 2 seems more popular based on GitHub stars, but Option 1 seems more active and has a possibly better API.

I'd like to try re-implementing this PR with jacobsa/fuse to see if it's objectively better than go-fuse.

Reply:
Reading through https://github.com/jacobsa/fuse/issues/78, I think for now we're good to stick with go-fuse. The fuse interface isn't exported, so we're in a position to rewrite the implementation in the future.